src/bindings/python/docs/code_examples.md

```diff
@@ -0,0 +1,414 @@
+
+Here is a simple blueprint for `MyTensor` class, notice how **not all** requirements are met here:
+```cpp
+class MyTensor {
+public:
+    // Constructing MyTensor is done with already created Tensor
+    MyTensor(ov::Tensor& t) : _tensor(t) {};
+
+    ~MyTensor() = default;
+
+    // Gets size from Tensor and return it
+    size_t get_size() const {
+        return _tensor.get_size();
+    }
+
+    // Public member that allows all operations on Tensor
+    ov::Tensor _tensor;
+};
+```
+
+Let's start binding code, align class name with C++ counterpart and add documentation string: 
+```cpp
+// Add class to the module
+py::class_<MyTensor, std::shared_ptr<MyTensor>> cls(mymodule, "MyTensor");
+
+cls.doc() = "These are my first class bindings!"
+```
+
+Create `__init__` and functions for the class. `py::arg` stand for actual arguments of the given function, remember these should match their C++ equivalent in order and count <!-- number? -->. However argument names are not required to be an exact copy of C++ ones, as different restricted keywords or shortnames could appear, pick them with a taste. :) Very unusual `R"(...)"` construct adds documentation string to the function. OpenVINO:tm: project follows *reST (reStructuredText)* format of docstrings.
+```cpp
+// This initialize use already implemented C++ constructor 
+cls.def(py::init<ov::Tensor&>(),
+        py::arg("tensor"),
+        R"(
+            MyTensor's constructor.
+
+            :param tensor: `Tensor` to create new `MyTensor` from.
+            :type tensor: openvino.runtime.Tensor
+        )");
+
+// This initialize use custom constructor, implemented via lambda 
+cls.def(py::init([](std::vector<float>& list) {
+                auto tensor = ov::Tensor(ov::element::f32, ov::Shape({list.size()}));
+                std::memcpy(tensor.data(), &list[0], list.size() * sizeof(float));
+                return MyTensor(tensor);
+            }),
+        py::arg("list"),
+        R"(
+            MyTensor's constructor.
+
+            :param list: List to create new `MyTensor` from.
+            :type list: list
+        )");
+
+// Binds class function directly
+cls.def("get_size", &MyTensor::get_size,
+        R"(
+            Get MyTensor's size.
+        )");
+
+// Adds function only on pybind's layer -- this function will become exclusive to Python API
+cls.def("say_hello", [](const MyTensor& self) {
+            py::print("Hello there!");
+            for (size_t i = 0; i < self.get_size(); i++) {
+                py::print(self._tensor.data<float>() + i);
+            }
+        },
+        R"(
+            Say hello and print contents of Tensor.
+        )");
+```
+
+*Tip: To add a function on the module's level as a "free function", simply define it on the module's object:*
+```cpp
+mymodule.def("get_smile", []() {
+    py::print(":)");
+});
+```
+
+**Don't forget to rebuild the project** and test it out:
```

Comment:
Would being explicit about rebuilding AND reinstalling the project (`make install` is required for Python changes to be applied) be beneficial? Without `make install` the contributor could get confused, not noticing new behavior in Python.

Reply:
Yeah... actually it will depend on `BUILDING.md` instructions, so I am going to keep this comment unresolved and get back to it as soon as Building Guide PR starts.